client/cmd/job/validate.go

```diff
@@ -39,109 +45,319 @@ func NewValidateCommand() *cobra.Command {
 
-	cmd.Flags().BoolVarP(&validate.verbose, "verbose", "v", false, "Print details related to operation")
-	cmd.Flags().StringVarP(&validate.namespaceName, "namespace", "n", validate.namespaceName, "Namespace of the resource within project")
 	cmd.MarkFlagRequired("namespace")
+
 	return cmd
 }
 
-func (v *validateCommand) PreRunE(_ *cobra.Command, _ []string) error { // Load mandatory config
+func (v *validateCommand) PreRunE(_ *cobra.Command, _ []string) error {
 	conf, err := config.LoadClientConfig(v.configFilePath)
 	if err != nil {
 		return err
 	}
 	v.clientConfig = conf
 
-	v.connection = connection.NewInsecure(v.logger)
+	v.connection = connection.New(v.logger, conf)
 	return nil
 }
 
 func (v *validateCommand) RunE(_ *cobra.Command, _ []string) error {
+	if v.forDelete && !v.fromServer {
+		return errors.New("for-delete flag is only valid with from-server flag")
+	}
+
+	if v.fromServer && len(v.jobNames) == 0 {
+		return errors.New("from-server flag is only valid with jobs flag being set")
+	}
+
 	namespace, err := v.clientConfig.GetNamespaceByName(v.namespaceName)
 	if err != nil {
 		return err
 	}
 
-	jobSpecReadWriter, err := specio.NewJobSpecReadWriter(afero.NewOsFs(), specio.WithJobSpecParentReading())
-	if err != nil {
+	v.logger.Info("Validating job specifications for namespace [%s]\n", v.namespaceName)
+	start := time.Now()
+
+	if err := v.executeLocalValidation(); err != nil {
+		v.logger.Info("Validation is finished, took %s", time.Since(start).Round(time.Second))
 		return err
 	}
 
-	start := time.Now()
-	projectName := v.clientConfig.Project.Name
-	v.logger.Info("Validating job specifications for project: %s, namespace: %s", projectName, namespace.Name)
-	jobSpecs, err := jobSpecReadWriter.ReadAll(namespace.Job.Path)
+	if err := v.executeServerValidation(namespace); err != nil {
+		v.logger.Info("Validation is finished, took %s", time.Since(start).Round(time.Second))
+		return err
+	}
+
+	v.logger.Info("Validation is finished, took %s", time.Since(start).Round(time.Second))
+	return nil
+}
+
+func (v *validateCommand) executeLocalValidation() error {
+	jobSpecReadWriter, err := specio.NewJobSpecReadWriter(afero.NewOsFs())
 	if err != nil {
-		return fmt.Errorf("directory '%s': %w", namespace.Job.Path, err)
+		return err
+	}
+
+	namespaceNamesByJobName := make(map[string][]string)
+	for _, namespace := range v.clientConfig.Namespaces {
+		jobs, err := jobSpecReadWriter.ReadAll(namespace.Job.Path)
+		if err != nil {
+			return err
+		}
+
+		for _, j := range jobs {
+			namespaceNamesByJobName[j.Name] = append(namespaceNamesByJobName[j.Name], namespace.Name)
+		}
+	}
+
+	success := true
+	for jobName, namespaceNames := range namespaceNamesByJobName {
+		if len(namespaceNames) == 1 {
+			continue
+		}
+
+		uniqueNames := v.deduplicate(namespaceNames)
+		v.logger.Error("[%s] is written [%d] times in namespace [%s]", jobName, len(namespaceNames), strings.Join(uniqueNames, ", "))
```

Comment:
the duplication is within a project in the local. so, if there is duplication in the same namespace, it will be detected as well. I can show the result in a different discussion.

Reply:
discussed separately, resolving it